### PR TITLE
Add Lockbox (DO NOT MERGE TILL AFTER 6/28 deploy and l10n merge)

### PR DESCRIPTION
**Pull request by devinreams**
Fixes https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/47

My bad, I gave bad advice so re-created @nickbrandt's branch here. Replacing #3623

- [x] We want to ensure that this is set to not be live for now.
- [x] We are unsure what ID # should be used for Lockbox in the YAML file
- [x] We need to get all the proper PI/QA folks included
- [x] Some other content may still need adjusted for other areas of the Lockbox Test Pilot instance.
- [x] but make sure it becomes "Featured" when ready to launch
- [x] Final items: https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/44

### CODE REVIEW

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/2a69b651cd3440fd1466a9fbb4cfc2f8178caf4c/

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: Should this be https://lockbox.firefox.com ?

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: Should this be /lockbox/ or /@lockbox/ or something instead of "experiment-name"?

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: Not sure if we want to merge w/ "dev: true" so we can test on the dev site without it risking riding the trains into production.

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: Feel free to remove this 🤡.
I can think of several other QA people far more deserving of credit than this donkey.

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: I think a lot of the following items are placeholders and may need some :eyes: before merging.
Like, we probably don't want `eol_warning` or `incompatible` here.

File: locales/en-US/experiments.ftl
Comment (pdehaan) [inline locales/en-US/experiments.ftl]: If you ✂️me from the `contributors:` list above (and please do), you'll also need to make sure you tweak these values, otherwise all the names+titles will get out of sync (unless you replace me with MBrandt -- although I'm guessing he has a much more glamorous job title than I).

File: content-src/experiments/color.yaml
Comment (pdehaan) [inline content-src/experiments/color.yaml:4]: We may not want to unfeature this yet.
I did make a note about adding `dev: true` to the lockbox.yaml below though, so feel free to ignore this comment temporarily until you read that. At which point you're welcome to ignore this comment permanantly.

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [reply content-src/experiments/lockbox.yaml]: Although, if you're an iOS app without an add-on, you may be able to drop these values.
You may want to check what Send did in https://github.com/mozilla/testpilot/blob/master/content-src/experiments/send.yaml

File: locales/en-US/experiments.ftl
Comment (flodolo) [inline locales/en-US/experiments.ftl]: HTML markup shouldn't really be in the string. Can this be split up in paragraphs?

Also the actual URL shouldn't be in the localized string.

File: locales/en-US/experiments.ftl
Comment (flodolo) [inline locales/en-US/experiments.ftl]: (non English speaker) Filter "on"?

File: locales/en-US/experiments.ftl
Comment (flodolo) [inline locales/en-US/experiments.ftl]: Uhm… what?

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: We also may want something a bit more... future-esque.

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/06223b8bb2fc9f2a7b58de5ffe1ac61b0ec78210/

File: locales/en-US/experiments.ftl
Comment (pdehaan) [reply locales/en-US/experiments.ftl]: I think this is just a copy of the placeholder text from https://github.com/mozilla/testpilot/pull/3624/files#diff-764620a2f7d2d9a3655e4a1806216a95R110

File: locales/en-US/experiments.ftl
Comment (pdehaan) [reply locales/en-US/experiments.ftl]: (poor English speaker) Yeah, I think "on" is unnecessary word here and above in https://github.com/mozilla/testpilot/pull/3624/files#diff-764620a2f7d2d9a3655e4a1806216a95R46

File: locales/en-US/experiments.ftl
Comment (pdehaan) [reply locales/en-US/experiments.ftl]: Sadly, I think we have prior art in excessive HTML markup in introductions in [at least]: min-vid, pageshot, and send:

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L64

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L114

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L130

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/b357c4942f09572d1ff88cc2ba0f4a2f9c57b3f9/

**Comment by lmorchard** [comment]
FWIW, [this failed](https://circleci.com/gh/mozilla/testpilot/11102?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) on the check for up-to-date strings from YAML content. [Probably needs an `npm run l10n:extract`](https://github.com/mozilla/testpilot/blob/master/docs/content/reference.md#localization) to update the `.ftl` files.

**Comment by devinreams** [comment]
>Probably needs an npm run l10n:extract to update the .ftl files.

@lmorchard Yeah, what's neat is when I change to `dev: true` then the l10n stuff gets skipped (and actually drops the lines from the experiments.ftl file.

~I'll poke around and see if there's something else that needs to change, but figured I'd mention it here before I dig too deep...~

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/frontend/lib/content/l10n.js#L11-L12

Removing strings for now to match expectations, will get re-added back when we switch dev to false.

File: locales/en-US/experiments.ftl
Comment (devinreams) [reply locales/en-US/experiments.ftl]: >HTML markup shouldn't really be in the string

Ah yes, this appears to be the result of the l10n extraction. I'd have to find a clever way to redo the YAML so the "introduction" could handle these strings and still be formatted as expected and that's probably beyond my ability and scope of this PR.

File: content-src/experiments/color.yaml
Comment (devinreams) [reply content-src/experiments/color.yaml:4]: >We may not want to unfeature this yet.

Good point, it was good to flip for testing but needs to remain in place before merging. 👍

File: content-src/experiments/lockbox.yaml
Comment (devinreams) [reply content-src/experiments/lockbox.yaml]: > can think of several other QA people far more deserving of credit

Agreed. I've added to our checklist above to finalize our contributor list.

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/cf80d252a23abeaeb06cc49361fd0dd33a1690aa/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/6c85519c9e895db50f9925940ba044535fad22ef/

**Comment by lmorchard** [comment]
> Yeah, what's neat is when I change to dev: true then the l10n stuff gets skipped (and actually drops the lines from the experiments.ftl file.

Oh, yeah, we don't extract strings for `dev: true` experiments - because we don't want them ending up in front of localizers when the experiments are still in relative flux and strings might be actively changing.

Probably a thing worth filing an issue to add to docs

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/c523a58de2729e10c759bc4c842e24b32a543256/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/3df6fe7cc1409708f10a7e040c1894d2371ffa99/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/d06d310d62bd5edcdedd39f282efc8b4df12cc13/

**Comment by devinreams** [comment]
OK this PR is ready for a final review.

One thing I'm thinking about is how to make the measurements section a bit more conditional so we can have 0 "measurements" defined in the yaml but have a privacy policy link appear with minimal code changes.

There are some 'go live' todos and final assets to pull together (in a follow up PR), those are captured here: https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/44

**Comment by flodolo** [comment]
> OK this PR is ready for a final review.

At this point it's missing changes to the localizable file though.

**Comment by devinreams** [comment]
>At this point it's missing changes to the localizable file though.

@flodolo Yeah my thinking was keep this in "dev: true" mode while we keep pulling things together and _then_ do a follow up PR which will include the final strings. I expect some changes from marketing and copy folks in the coming weeks...

**Comment by flodolo** [comment]
> @flodolo Yeah my thinking was keep this in "dev: true" mode while we keep pulling things together and then do a follow up PR which will include the final strings. I expect some changes from marketing and copy folks in the coming weeks...

OK, makes sense then.

File: content-src/experiments/lockbox.yaml
Comment (pdehaan) [inline content-src/experiments/lockbox.yaml]: I haven't looked into TestPilot schemas lately to see if `measurements` is required, but I'll just call this out because it is literally telling me to remove it.

Comment (devinreams) [reply content-src/experiments/lockbox.yaml]: Yeah the page dies without something here. Even though it pulls the localized string (not this) into the page itself.

I want to see about making measures _not_ required and not displaying the bulleted list in "Your privacy" section if empty in a follow-up PR.

**Comment by devinreams** [comment]
@meandavejustice are you OK with merging this as-is (`dev: true` so should be safe)?

I'm thinking from here:

- soon: merge this as-is
- next week: [get the final assets in](https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/44)
- June 28th: tag the release
- July 3: push to prod
- July 10: live/launch

Does that seem right to you?

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/23fe9b44df8bf0cfaf69f73be0575fe1f12f12b8/

**Comment by johngruen** [comment]
Hey this is looking good. Let's hold off on merging for a sec for reasons i'll explain in a pending bug

**Review by meandavejustice — APPROVED** [review]

**Comment by devinreams** [comment]
> Let's hold off on merging for a sec for reasons i'll explain in a pending bug

OK, holding off.   👍  

Got a few more minor changes landed in the meantime. Seems like we may be able to get this whole branch finished up in the next day or so..

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/c9da90fb6d8dd6a932a3fb3f9b7c161c4dd4bf2d/

**Comment by devinreams** [comment]
@flodolo with the revised deploy schedule from @johngruen we are going to put _all_ the Lockbox experiment assets and strings in this branch and push it to master after the tag this week (so on Friday this week).

This way translators _could_ start localizing this experiment (although it will initially be available to select EN countries, only).

I believe I addressed all of your initial comments and concerns in this PR. Would you mind taking another look to make sure these strings are OK (given the limitation of _needing_ HTML in some of them based on how the page's output is built).

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/3756d8bc0401eff5d3dc7f2ef415494ea4e4c743/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/444ea30f7fa8d01770901ad5ee7aeefff0b1d1a6/

**Comment by devinreams** [comment]
For those following along at home:

screenshot_2018-06-26 firefox test pilot 2
screenshot_2018-06-26 firefox test pilot 3
screenshot_2018-06-26 firefox test pilot 4

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/dfd8b414786ce656b66bd37f8bd7202abc786947/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/c2d40e48e7507e7ac40440bea60fce688f7d67b2/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/ff7bfcf649cc4bcbe3b97d27aa83b6d201fb6b37/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/d5c5f297e0642d9ef00dc104e5da0822d09cf6b8/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/d5c5f297e0642d9ef00dc104e5da0822d09cf6b8/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3624/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/35d07f8194783a6794bcc356be787c46a6b3cbf2/

**Comment by meandavejustice** [comment]
@devinreams Don't worry about the integration test failing here, we filed #3476 as a follow up issue because of some addon environment discrepancies.

**Comment by devinreams** [comment]
Thanks, @meandavejustice. I'll merge now then since the release has been tagged. 👌

**Pull request merged**
